src/lib.rs

```diff
@@ -1,6 +1,343 @@
+    }
+
+    /// Creates a new client with custom flags
+    pub fn new_custom(name: &str, flags: MMVFlags) -> io::Result<Client> {
+        let cluster_id =
+            rand::thread_rng().gen::<u32>() & ((1 << CLUSTER_ID_BIT_LEN) - 1);
+
+        let mut mmv_path = get_mmv_dir()?;
+        mmv_path.push(name);
+
+        Ok(Client {
+            flags: flags,
+            cluster_id: cluster_id,
+            mmv_path: mmv_path
+        })
+    }
+
+    /// Exports metrics by writing to an MMV file
+    pub fn export(&self) -> io::Result<()> {
+        let mut file = OpenOptions::new().read(true).write(true).create(true)
+            .open(&self.mmv_path)?;
+        let mmv_size = HDR_LEN + 2*TOC_BLOCK_LEN;
+        file.write(&vec![0; mmv_size])?;
+
+        let mut mmap = Mmap::open(&file, Protection::ReadWrite)?;
+        let mut cursor = Cursor::new(unsafe { mmap.as_mut_slice() });
+
+        let writer_info = self.write_mmv_header(&mut cursor)?;
+        self.write_metric_toc_block(&mut cursor)?;
+        self.write_values_toc_block(&mut cursor)?;
+        self.unlock_mmv_header(&mut cursor, &writer_info)
+    }
+
+    fn write_mmv_header(&self, cursor: &mut Cursor<&mut [u8]>)
+        -> io::Result<MMVWriterInfo> {
+        let mut writer_info = MMVWriterInfo {
+            gen: 0,
+            gen2off: 0
+        };
+    
+        // MMV\0
+        cursor.write_all(CString::new("MMV")?.to_bytes_with_nul())?;
+        // version
+        cursor.write_u32::<Endian>(1)?;
+        // generation1
+        let gen = time::now().to_timespec().sec;
+        cursor.write_i64::<Endian>(gen)?;
+        writer_info.gen = gen;
+        // generation2
+        writer_info.gen2off = cursor.position();
+        cursor.write_i64::<Endian>(0)?;
+        // no. of toc blocks
+        cursor.write_i32::<Endian>(2)?;
+        // flags
+        cursor.write_u32::<Endian>(self.flags.bits())?;
+        // pid
+        cursor.write_i32::<Endian>(get_process_id())?;
+        // cluster id
+        cursor.write_u32::<Endian>(self.cluster_id)?;
+
+        Ok(writer_info)
+    }
+
+    fn write_metric_toc_block(&self, cursor: &mut Cursor<&mut [u8]>)
+         -> io::Result<()> {
+        // section type
+        cursor.write_u32::<Endian>(3)?;
+        // no. of entries
+        cursor.write_u32::<Endian>(0)?;
+        // section offset
+        cursor.write_u64::<Endian>(0)
+    }
+
+    fn write_values_toc_block(&self, cursor: &mut Cursor<&mut [u8]>)
+         -> io::Result<()> {
+        // section type
+        cursor.write_u32::<Endian>(4)?;
+        // no. of entries
+        cursor.write_u32::<Endian>(0)?;
+        // section offset
+        cursor.write_u64::<Endian>(0)
+    }
+
+    fn unlock_mmv_header(&self, cursor: &mut Cursor<&mut [u8]>,
+        writer_info: &MMVWriterInfo)-> io::Result<()> {
+        cursor.set_position(writer_info.gen2off);
+        cursor.write_i64::<Endian>(writer_info.gen)
+    }
+
+    pub fn cluster_id(&self) -> u32 {
```

Comment:
@natoscott I see. Though the issue remains of whether we need to expose it publicly as a part of hornet's instrumentation API. If we don't, it'll still be accessible through the mmvdump module I'll implement later on.